framework/tst/dslabs/framework/testing/visualization/VizConfig.java

```diff
@@ -45,12 +45,26 @@
  */
 public abstract class VizConfig {
     public SearchState getInitialState(String[] args) {
-        return getInitialState(Integer.parseInt(args[0]),
-                Integer.parseInt(args[1]), commands(args[2]));
+        int numServers = Integer.parseInt(args[0]);
+        int numClients = Integer.parseInt(args[1]);
+        // The input is in one of the following formats:
+        //
+        // <# servers> <# clients> <uniform workload>
+        //
+        // <# servers> <# clients> <client 1 workload> ... <client n workload>
+        if (args.length != 3 && args.length != 2 + numClients) {
+            throw new IllegalArgumentException("Please provide either a single workload for all " +
+                                               "clients or a separate workload for each client.");
+        }
+        List<List<String>> commands = new LinkedList<>();
+        for (int i = 2; i < args.length; i++) {
```

Comment:
See comment on `PingVizConfig`

Reply:
Fixed in [f69e1cb](https://github.com/emichael/dslabs/pull/19/commits/f69e1cbb87f3fe5ce08bc8c635e9e590aae4d934).